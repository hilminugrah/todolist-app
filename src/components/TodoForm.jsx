import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addTodo } from '../redux/actions/todoActions';

function TodoForm() {
  const [title, setTitle] = useState('');
  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();
    if (title.trim() === '') return;

    dispatch(addTodo(title));
    setTitle('');
  };

  return (
    <form onSubmit={handleSubmit} className="flex items-center gap-2 mb-6 w-full max-w-xl mx-auto mt-8">
      <input type="text" value={title} onChange={(e) => setTitle(e.target.value)} placeholder="Tambah tugas baru" className="flex-1 border border-gray-500 rounded px-3 py-2 shadow-sm focus:outline-none" />
      <button type="submit" className="bg-blue-900 text-white px-4 py-2 rounded hover:bg-blue-500 transition flex items-center gap-2">
        <span>Tambah</span>
        <span className="text-lg">+</span>
      </button>
    </form>
  );
}

export default TodoForm;
