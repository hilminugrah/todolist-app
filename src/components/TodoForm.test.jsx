import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import TodoForm from './TodoForm';

const mockStore = configureStore([]);

describe('TodoForm Component', () => {
  let store;
  let mockDispatch;

  beforeEach(() => {
    store = mockStore({});
    mockDispatch = jest.fn();
    store.dispatch = mockDispatch;
  });

  it('sesuai snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TodoForm />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('tidak memanggil dispatch jika input kosong', () => {
    render(
      <Provider store={store}>
        <TodoForm />
      </Provider>
    );

    const submitButton = screen.getByRole('button', { name: /Tambah \+/i });
    fireEvent.click(submitButton);

    expect(mockDispatch).not.toHaveBeenCalled();
  });

  it('memanggil dispatch dan reset input jika ada teks', () => {
    render(
      <Provider store={store}>
        <TodoForm />
      </Provider>
    );

    const input = screen.getByPlaceholderText(/Tambah tugas baru/i);
    const submitButton = screen.getByRole('button', { name: /Tambah \+/i });

    fireEvent.change(input, { target: { value: 'Belajar Redux' } });
    fireEvent.click(submitButton);

    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(input.value).toBe('');
  });
});
