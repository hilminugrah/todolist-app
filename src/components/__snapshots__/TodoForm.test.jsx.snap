// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TodoForm Component sesuai snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex items-center gap-2 mb-6 w-full max-w-xl mx-auto mt-8"
  >
    <input
      class="flex-1 border border-gray-500 rounded px-3 py-2 shadow-sm focus:outline-none"
      placeholder="Tambah tugas baru"
      type="text"
      value=""
    />
    <button
      class="bg-blue-900 text-white px-4 py-2 rounded hover:bg-blue-500 transition flex items-center gap-2"
      type="submit"
    >
      <span>
        Tambah
      </span>
      <span
        class="text-lg"
      >
        +
      </span>
    </button>
  </form>
</DocumentFragment>
`;
